src/mappings/event/loans.ts

```diff
@@ -308,16 +322,16 @@ export async function depositCollateral(
         symbol = await symbolFromCurrency(newCurrency);
         const qRate = cachedRates.getRate(block.height, symbol);
 
-        const newAmount = Number(amount) * qRate.rate;
+        const newAmount = Big(amount.toString()).mul(qRate.rate);
         amounts = await getExchangeRate(ctx, block.timestamp, newCurrency, newAmount);
         symbol = `q`.concat(symbol);
         if(newCurrency){
             comment = `${getFirstAndLastFour(account)} deposited ${await friendlyAmount(newCurrency, newAmount)} for collateral`
         }
     } else {
         symbol = await symbolFromCurrency(currency);
-        amounts = await getExchangeRate(ctx, block.timestamp, currency, Number(amount));
-        comment = `${getFirstAndLastFour(account)} deposited ${await friendlyAmount(currency, Number(amount))} for collateral`
+        amounts = await getExchangeRate(ctx, block.timestamp, currency, amount.toString());
+        comment = `${getFirstAndLastFour(account)} deposited ${await friendlyAmount(currency, amount.toString())} for collateral`
     }
```

Comment:
I didn't bother to check/change any logic beyond fixing the type conversions.

I suppose that could be unreachable, but prefer to leave it as is for now rather than introduce the risk of missing some edge case I'm not aware of.